runtime/tests/org.jboss.tools.runtime.test/src/org/jboss/tools/runtime/handlers/TestHandler2.java

```diff
@@ -0,0 +1,94 @@
+/*******************************************************************************
+  * Copyright (c) 2010 - 2015 Red Hat, Inc.
+  * Distributed under license by Red Hat, Inc. All rights reserved.
+  * This program is made available under the terms of the
+  * Eclipse Public License v1.0 which accompanies this distribution,
+  * and is available at http://www.eclipse.org/legal/epl-v10.html
+  *
+  * Contributor:
+  *     Red Hat, Inc. - initial API and implementation
+  ******************************************************************************/
+package org.jboss.tools.runtime.handlers;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.Path;
+import org.jboss.tools.runtime.core.model.AbstractRuntimeDetectorDelegate;
+import org.jboss.tools.runtime.core.model.RuntimeDefinition;
+import org.jboss.tools.runtime.test.RuntimeTestActivator;
+
+public class TestHandler2 extends AbstractRuntimeDetectorDelegate {
+
+	private static TestHandler2 ME;
+	
+	public static TestHandler2 getInstance() {
+		return ME;
+	}
+	
+	public TestHandler2() {
+		ME = this;
+		System.out.println("Inside TestHandler2 constructor: " + this.toString());
+		try {
+			throw new Exception();
+		} catch(Exception e) {
+			e.printStackTrace();
```

Comment:
Why do need to print the stack trace in the constructor? And btw Thread.dumpStack(); would look cleaner I guess ;-)